### ISSUE TITLE
Install instructions

### ISSUE THREAD

**Issue by magician11** [ssbc/patchbay#36]
I ran

`npm install scuttlebot@latest -g`

with no problems.

However the next step on checking for secure-scuttlebutt..

```
~/Desktop/golightly+ » npm ls secure-scuttlebutt -g
/Users/andrewgolightly/.nvm/versions/node/v6.2.2/lib
└── (empty)

npm ERR! code 1

```

Then I tried..

```
~/Desktop/golightly+ » npm ls -g
/Users/andrewgolightly/.nvm/versions/node/v6.2.2/lib
└── (empty)

```

I do see modules installed if I list the contents of that directory though..

```
~/Desktop/golightly+ » ll ~/.nvm/versions/node/v6.2.2/lib/node_modules
total 8
drwxr-xr-x  24 andrewgolightly  staff   816B 17 Jun 21:34 npm
drwxr-xr-x   8 andrewgolightly  staff   272B 24 Jun 23:23 npm-check-updates
drwxr-xr-x  14 andrewgolightly  staff   476B 13 Jul 08:50 patchwork
drwxr-xr-x  14 andrewgolightly  staff   476B 13 Jul 08:40 scuttlebot
lrwxr-xr-x   1 andrewgolightly  staff    47B 21 Jun 16:41 ssb-patchwork -> /Users/andrewgolightly/work/bitnation/patchwork
```

The after running `sbot server`, in a new tab I ran..

```
~ » sbot plugins.install ssb-links ssb-query
TypeError: Param 1 must by of type object
Install a plugin to Scuttlebot.

install {nodeModule} [--from path]

Calls out to npm to install a package into
`~/.ssb/node_modules`.

-   nodeModule (string): The name of the plugin to install.
Uses npm's module package-name rules.
-   from (string): A location to install from (directory path,
url, or any location that npm accepts for its install
command).
```

So I thought it had something to do with patchwork not being at the latest version.. so I tried

```
~ » npm install ssb-patchwork -g
npm ERR! Darwin 14.5.0
npm ERR! argv "/Users/andrewgolightly/.nvm/versions/node/v6.2.2/bin/node" "/Users/andrewgolightly/.nvm/versions/node/v6.2.2/bin/npm" "install" "ssb-patchwork" "-g"
npm ERR! node v6.2.2
npm ERR! npm  v3.9.5
npm ERR! path /Users/andrewgolightly/.nvm/versions/node/v6.2.2/lib/node_modules/ssb-patchwork
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall realpath

npm ERR! enoent ENOENT: no such file or directory, realpath '/Users/andrewgolightly/.nvm/versions/node/v6.2.2/lib/node_modules/ssb-patchwork'
npm ERR! enoent ENOENT: no such file or directory, realpath '/Users/andrewgolightly/.nvm/versions/node/v6.2.2/lib/node_modules/ssb-patchwork'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/andrewgolightly/npm-debug.log
```

.. and then gave up.

**Comment by evbogue** [comment]
@magician11 

For the sbot plugins step, the install instructions in the readme are wrong. plugins.install only accepts on argument at a time. Try

```
sbot plugins.install ssb-query
sbot plugins.install ssb-links
```

I'm not sure what is going wrong with your global installs though. Maybe nvm is the issue? 

You could try cloning patchwork using git instead of using npm to install it.

**Comment by magician11** [comment]
@evbogue 

Thanks, that worked. That got me a step closer.

I used git to install patchwork as you suggested.

How do I modify the line `patchwork plugins.install ssb-links ssb-query` then to use the patchwork that is installed to a specific directory?

**Comment by evbogue** [comment]
@magician11 The readme has been updated, and in my experience the sbot either/or patchwork is correct. plugins.install basically runs an npm install in your ~/.ssb folder, so you don't need to do it twice and can use patchwork plugins.install or sbot plugins.install.

Once you have patchwork installed you can start it by typing

**Comment by magician11** [comment]
Ok, I've got patchwork running fine. A little confused on the focus on patchwork, because patchbay is a replacement for it?

When I went through the patchbay instructions in the readme.. 

In one tab I have sbot server running with the installed plugins, and then in another tab I run electro in the patchbay directory, and electron opens and it's blank.

**Comment by evbogue** [comment]
@magician11 Well, you shouldn't be seeing blank. Did you use the command

```
% CHLORIDE_JS=1 electro index.js
```

to start patchbay? See issue #6 for why this is necessary.

Patchbay was bootstrapped off Patchwork, and dependence is being phased out. Right now it only depends on Patchwork for images -- because no one has got around to serving ssb-blobs yet.

I've used git-ssb-web to serve images for Patchbay instead, you can either change the port number in the Patchbay code or you can change git-ssb-web's port number to port 7777.

**Comment by evbogue** [comment]
@magician11 Would you be able try the Patchbay installation again from scratch? @dominictarr has eliminated the Patchwork dependency, the CHLORIDE_JS=1 problem, and simplified the installation instructions.

See: https://github.com/dominictarr/patchbay/blob/master/README.md

I'm also curious if you think these installation instructions qualify as 'easy', and if you think we can close #14.

**Comment by magician11** [comment]
Ok, I started from scratch and it all installed no prob.

But the final step where it opened electron, it still just displays a blank window.

It was pretty easy to go through with no issues though.

**Comment by dominictarr** [comment]
@magician11 are you running a `sbot server` in another tab? you should see some ui stuff though.
is there any errors in the devtools? (ctrl-shift-i)

**Comment by magician11** [comment]
@dominictarr yes, I'm running sbot server. This is what I see...

![screen shot 2016-07-22 at 3 40 55 pm](https://cloud.githubusercontent.com/assets/3735849/17051422/dc8a6772-5022-11e6-9b89-0fd14d5e99ed.png)

**Comment by dominictarr** [comment]
@magician11 I see a "cannot find module 'hyperscript'" which makes me think you just need to do `npm install` in the patchbay directory?

you are running `electro index.js` right? (not electron)

**Comment by magician11** [comment]
Correct, I'm running `electro index.js`

I installed that missing module... and then when I ran it again, it said it was missing `pull-stream`, so I installed that.. then it said it was missing `pull-next` .. and then I stopped. Not sure how many times I'll need to keep installing modules?

**Comment by evbogue** [comment]
@magician11 I believe the readme is missing the most taken-for-granted step, the `npm install` inside of the project folder.

```
git clone https://github.com/dominictarr/patchbay.git
cd patchbay
npm install 
npm install electro electron-prebuilt -g
electro index.js
```

This is why you're seeing missing dependencies.

**Comment by dominictarr** [comment]
@evbogue is right!

**Comment by dominictarr** [comment]
fixed in 1.8.5

**Comment by magician11** [comment]
Yes! I'm really good at following steps exactly... nice catch @evbogue
